File: Core/src/main/java/com/hqq/core/utils/sp/SharedPreferencesUtil.java
package com.hqq.core.utils.sp;

import android.content.Context;
import android.content.SharedPreferences;

public class SharedPreferencesUtil {
    public static synchronized void  putObject(Context context, String key, Object object) {
        //PreferenceManager.getDefaultSharedPreferences(context);
        SharedPreferences sp = context.getSharedPreferences("preferences", Context.MODE_PRIVATE |Context.MODE_MULTI_PROCESS);
        SharedPreferences.Editor editor = sp.edit();
        if (object instanceof Boolean) {
            editor.putBoolean(key, (Boolean) object);
        } else if (object instanceof Float) {
            editor.putFloat(key, (Float) object);
        } else if (object instanceof Integer) {
            editor.putInt(key, (Integer) object);
        } else if (object instanceof Long) {
            editor.putLong(key, (Long) object);
        } else if (object instanceof String) {
            editor.putString(key, (String) object);
        } else {
            editor.putString(key, object.toString());
        }
        editor.apply();
    }

    public static synchronized Object getObject(Context context, String key, Object defaultObject) {
        try {
            //  PreferenceManager.getDefaultSharedPreferences(context); // 会出现空指针=
            SharedPreferences sp = context.getSharedPreferences("preferences", Context.MODE_PRIVATE |Context.MODE_MULTI_PROCESS);
            if (defaultObject instanceof Boolean) {
                return sp.getBoolean(key, (Boolean) defaultObject);
            } else if (defaultObject instanceof Float) {
                return sp.getFloat(key, (Float) defaultObject);
            } else if (defaultObject instanceof Integer) {
                return sp.getInt(key, (Integer) defaultObject);
            } else if (defaultObject instanceof Long) {
                return sp.getLong(key, (Long) defaultObject);
            } else if (defaultObject instanceof String) {
                return sp.getString(key, (String) defaultObject);
            }
            return null;
        } catch (Exception e) {
            return null;
        }

    }

}

File: Core/src/main/java/com/hqq/core/utils/sp/SharedPreferencesUtil.kt
package com.hqq.core.utils.sp

import android.content.Context

/**
 * @Author : huangqiqiang
 * @Package : com.hqq.core.utils.sp
 * @FileName :   SharedPreferencesUtil.kt
 * @Date  : 2020/9/1 0001  下午 5:20
 * @Email :  qiqiang213@gmail.com
 * @Descrive :
 */
object SharedPreferencesUtil {
    @Synchronized
    fun putObject(context: Context, key: String?, `object`: Any) {
        //PreferenceManager.getDefaultSharedPreferences(context);
        val sp = context.applicationContext.getSharedPreferences("preferences", Context.MODE_PRIVATE or Context.MODE_MULTI_PROCESS)
        val editor = sp.edit()
        if (`object` is Boolean) {
            editor.putBoolean(key, `object`)
        } else if (`object` is Float) {
            editor.putFloat(key, `object`)
        } else if (`object` is Int) {
            editor.putInt(key, `object`)
        } else if (`object` is Long) {
            editor.putLong(key, `object`)
        } else if (`object` is String) {
            editor.putString(key, `object`)
        } else {
            editor.putString(key, `object`.toString())
        }
        editor.apply()
    }

    @Synchronized
    fun getObject(context: Context, key: String?, defaultObject: Any?): Any? {
        return try {
            //  PreferenceManager.getDefaultSharedPreferences(context); // 会出现空指针=
            val sp = context.applicationContext.getSharedPreferences("preferences", Context.MODE_PRIVATE or Context.MODE_MULTI_PROCESS)
            if (defaultObject is Boolean) {
                return sp.getBoolean(key, (defaultObject as Boolean?)!!)
            } else if (defaultObject is Float) {
                return sp.getFloat(key, (defaultObject as Float?)!!)
            } else if (defaultObject is Int) {
                return sp.getInt(key, (defaultObject as Int?)!!)
            } else if (defaultObject is Long) {
                return sp.getLong(key, (defaultObject as Long?)!!)
            } else if (defaultObject is String) {
                return sp.getString(key, defaultObject as String?)
            }
            null
        } catch (e: Exception) {
            null
        }
    }
}